staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -352,6 +356,125 @@ func TestCacherWatchSemanticInitialEventsExtended(t *testing.T) {
 	storagetesting.RunWatchSemanticInitialEventsExtended(context.TODO(), t, store)
 }
 
+func TestStreamSeparation(t *testing.T) {
```

Comment:
This test would better fit cacher_whitebox_test.go instead (in this file we're running a set of implementation-agnostic tests)

Reply:
Done